### PR TITLE
Update Terraform provider to include new opensearch acl changes

### PR DESCRIPTION
**Pull request by bhardwajRahul**
This PR aims at adding support for opensearch acl changes to DO's Terraform Provider.

##### Testing instructions(for future references):
```
make testacc TESTARGS='-run=TestAccDigitalOceanDatabaseUser_OpenSearchACLs'

```

### CODE REVIEW

**Comment by loosla** [comment]
Looks good to me :rocket: :rocket: :rocket: 

Just a small question (nothing critical, just curiosity):
Do you think that `normalizeOpenSearchPermission` is reachable with this configuration 
```
                      "permission": {
				Type:     schema.TypeString,
				Required: true,
				ValidateFunc: validation.StringInSlice([]string{
					"deny",
					"admin",
					"read",
					"write",
					"readwrite",
				}, false),
			},
```
It looks like anything that is not in a list of allowed values will be denied before we get to normalisation.
So, `read_write` will be denied as well as uppercase values like `READ`.
Normalisation still can make sense if _validation_ will be changed in the future though. :thinking:

**Comment by bhardwajRahul** [comment]
> Looks good to me 🚀 🚀 🚀
> 
> Just a small question (nothing critical, just curiosity): Do you think that `normalizeOpenSearchPermission` is reachable with this configuration
> 
> ```
>                       "permission": {
> 				Type:     schema.TypeString,
> 				Required: true,
> 				ValidateFunc: validation.StringInSlice([]string{
> 					"deny",
> 					"admin",
> 					"read",
> 					"write",
> 					"readwrite",
> 				}, false),
> 			},
> ```
> 
> It looks like anything that is not in a list of allowed values will be denied before we get to normalisation. So, `read_write` will be denied as well as uppercase values like `READ`. Normalisation still can make sense if _validation_ will be changed in the future though. 🤔

@loosla yes, this has been added for the pupose of validation and we are using similar pattern for kafka acls also.

File: digitalocean/database/resource_database_user_test.go [hunk 1]
```diff
@@ -498,6 +557,60 @@ resource "digitalocean_database_user" "foobar_user" {
   }
 }`
 
+const testAccCheckDigitalOceanDatabaseUserConfigOpenSearchACL = `
+resource "digitalocean_database_cluster" "foobar" {
+  name       = "%s"
+  engine     = "openserach"
```
Comment (andrewsomething) [inline digitalocean/database/resource_database_user_test.go:563]:
```suggestion
  engine     = "opensearch"
```

```
=== CONT  TestAccDigitalOceanDatabaseUser_OpenSearchACLs
    resource_database_user_test.go:267: Step 1/2 error: Error running apply: exit status 1
        
        Error: Error creating database cluster: POST https://api.digitalocean.com/v2/databases: 422 (request "f05bf517-5041-4a70-b64a-a80330bcb550") invalid engine
        
          with digitalocean_database_cluster.foobar,
          on terraform_plugin_test.tf line 2, in resource "digitalocean_database_cluster" "foobar":
           2: resource "digitalocean_database_cluster" "foobar" {
        
--- FAIL: TestAccDigitalOceanDatabaseUser_OpenSearchACLs (2.70s)
```

**Review by andrewsomething — CHANGES_REQUESTED** [review]
Looks like a typo in the acceptance test config is causing a failure.

**Review by loosla — APPROVED** [review]
:+1:

File: digitalocean/database/resource_database_user_test.go [hunk 2]
```diff
@@ -498,6 +557,60 @@ resource "digitalocean_database_user" "foobar_user" {
   }
 }`
 
+const testAccCheckDigitalOceanDatabaseUserConfigOpenSearchACL = `
+resource "digitalocean_database_cluster" "foobar" {
+  name       = "%s"
+  engine     = "opensearch"
+  version    = "2"
+  size       = "db-s-2vcpu-2gb"
```
Comment (andrewsomething) [inline digitalocean/database/resource_database_user_test.go:565]: This doesn't seem to be a valid size for an OpenSearch cluster:

```
        Error: Error creating database cluster: POST https://api.digitalocean.com/v2/databases: 422 (request "822ac77c-4ce2-489f-9d3d-838e22475d6a") invalid size
        
          with digitalocean_database_cluster.foobar,
          on terraform_plugin_test.tf line 2, in resource "digitalocean_database_cluster" "foobar":
           2: resource "digitalocean_database_cluster" "foobar" {
```

Can you please ensure your new acceptance test passes. You can run just your test with:

```
make testacc TESTARGS='-run=TestAccDigitalOceanDatabaseUser_OpenSearchACLs'
```

**Review by andrewsomething — CHANGES_REQUESTED** [review]

**Comment by bhardwajRahul** [comment]
image

**Review by andrewsomething — APPROVED** [review]
:+1:  Thanks for fixing up the acceptance test! LGTM!

**Pull request merged**